integration-tests/tests/test_identity_transformation.py

```diff
@@ -0,0 +1,122 @@
+import shutil
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from typing import IO
+
+import pytest
+from tests.utils.config import (
+    BaseConfig,
+    DatasetLogs,
+)
+from tests.utils.utils import (
+    diff_equal,
+    run_and_assert,
+)
+
+pytestmark = pytest.mark.binaries
+
+text_datasets = pytest.mark.parametrize(
+    "download_and_extract_dataset",
+    [
+        "hive_24hr",
+    ],
+    indirect=["download_and_extract_dataset"],
+)
+
+json_datasets = pytest.mark.parametrize(
+    "download_and_extract_dataset",
+    [
+        "spark_event_logs",
+        "postgresql",
+    ],
+    indirect=["download_and_extract_dataset"],
+)
+
+
+@pytest.mark.clp
+@text_datasets
+def test_clp_identity_transform(
+    request, base_config: BaseConfig, download_and_extract_dataset: DatasetLogs
+) -> None:
+    binary_path_str = str(base_config.clp_bin_dir / "clp")
+    dataset_name = download_and_extract_dataset.name
+    download_dir = base_config.uncompressed_logs_dir / dataset_name
```

Comment:
From the ownership perspective, since dataset is downloaded an maintained by `DatasetLogs`, the `DatasetLogs` should maintain a download_dir variable itself.

Looking at this function, I think it is worthwhile changing download_dir to something like input_log_dir, or source_log_dir.